### ISSUE TITLE
Use profile credentials for aws s3 upload

### ISSUE THREAD

**Issue by reddyshop** [dealmore/terraform-aws-next-js#126]
Is it possible to allow using profile credentials for aws s3 upload? Currently it is required to specify AWS credentials as environment variables, and this overrides any profile configuration specified in the terraform file with workspaces.

**Comment by ofhouse** [comment]
Hi, there is an option called `use_awscli_for_static_upload` which uses the AWS CLI for uploading instead of the script that requires the environment variables.
AWS CLI must be installed for this to work.

Never tested this option but I think it currently only works if the default profile is used for uploading to the S3 bucket, since you cannot specify a profile that should be used here (but we surely could add this option if needed).

**Comment by reddyshop** [comment]
Thanks, I realized this same thing looking at the code and managed to get our build pipeline working using multiple profiles by also defining the default profile. In the context of build pipelines I'm thinking it might make sense to separate the terraform apply step from the actual pushing of static assets, to make the configuration easier to manage. Don't know if it is feasible to do here. Thank you very much for this project!

**Comment by ofhouse** [comment]
Yep, initially the publishing of the static assets was part of the `tf-next` CLI but we then decided to integrate it into the apply step, since the apply currently also manages the Lambdas that can vary with each apply.

But yes, Terraform is the wrong platform for continuous deployments like this.
I currently look into the possibility to use Terraform to setup the initial system and then use a tool like [Waypoint](https://www.waypointproject.io/) to manage the deployments to the platform then.

**Comment by ofhouse** [comment]
This is now available in [`v1.0.0-canary.2`](https://github.com/milliHQ/terraform-aws-next-js/releases/tag/v1.0.0-canary.2).
It now uses the `tf-next deploy` command which supports an optional `--awsProfile` flag:
```
tf-next deploy --awsProfile <custom-profile>
```
